rotkehlchen/chain/evm/decoding/superchain_bridge/decoder.py

```diff
@@ -13,64 +12,88 @@
 )
 from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
 from rotkehlchen.chain.evm.decoding.utils import bridge_match_transfer, bridge_prepare_data
-from rotkehlchen.chain.evm.types import string_to_evm_address
-from rotkehlchen.constants.assets import A_OPTIMISM_ETH
 from rotkehlchen.constants.resolver import evm_address_to_identifier
 from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
 from rotkehlchen.logging import RotkehlchenLogsAdapter
 from rotkehlchen.types import ChainID, ChecksumEvmAddress, EvmTokenKind
 from rotkehlchen.utils.misc import bytes_to_address
 
-BRIDGE_ADDRESSES: Final = (
-    string_to_evm_address('0x4200000000000000000000000000000000000010'),  # Normal Bridge
-    string_to_evm_address('0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65'),  # L2DAITokenBridge
-)
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
 
 DEPOSIT_FINALIZED: Final = b'\xb0DE#&\x87\x17\xa0&\x98\xbeG\xd0\x80:\xa7F\x8c\x00\xac\xbe\xd2\xf8\xbd\x93\xa0E\x9c\xdea\xdd\x89'  # noqa: E501
 WITHDRAWAL_INITIATED: Final = b"s\xd1p\x91\n\xba\x9emP\xb1\x02\xdbR+\x1d\xbc\xd7\x96!oQ(\xb4E\xaa!5'(\x86I~"  # noqa: E501
 
-
 logger = logging.getLogger(__name__)
 log = RotkehlchenLogsAdapter(logger)
 
 
-class OptimismBridgeDecoder(DecoderInterface):
-    def _decode_receive_deposit(self, context: DecoderContext) -> DecodingOutput:
-        """Decodes a bridging event. Either a deposit or a withdrawal
-        DAI uses special bridge. See https://github.com/makerdao/optimism-dai-bridge
+class SuperchainBridgeCommonDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
```

Comment:
not needed. This is already part of the evm_inquirer and perhaps even base_tools.

Reply:
no, it isn't present in `evm_inquirer` nor `base_tools`. plus, `DecoderInterface` has this signature:

https://github.com/prettyirrelevant/rotki/blob/ff8f9d93ce4c1e349414b99e108303df1f198bbc/rotkehlchen/chain/evm/decoding/interfaces.py#L81-L88
<img width="709" alt="image" src="https://github.com/user-attachments/assets/41553cc2-a9a8-4e02-a825-3fb49b6cdaaf">

Reply:
huh ... you are right. Interesting. We should probably think of this as repeating msg_aggregator everywhere here is silly.

It's still in evm_inquirer. Just `evm_inquirer.database.msg_aggregator`